src/v2/controllers/FormController.js

```diff
@@ -51,9 +50,8 @@ export default Controller.extend({
   invokeAction (actionPath = '') {
     const idx = this.options.appState.get('idx');
     if (idx['neededToProceed'].find(item => item.name === actionPath)) {
```

Comment:
I think this `if` and next `if` at line 64 can be part of one if..elseif..else clause.. What do you think?

Reply:
Could be. I guess it's about preference on code structure. ;)

However, I think it's a bit hacky (possible confusing) to invoke remediation Form in this function . This is probably quick work around today.
Initially, we would expect `invokeAction` only used to invoke form outside remediation and use `switchForm` to render other remediation forms.
Wonder we shall enhance `switchForm` function to see
- if the form has no ui schema, then just submit it.
- otherwise, render that form

thoughts ?
also @nikhilvenkatraman-okta

Reply:
yeah, I think ideally this function would only deal with invoking the action functions that are inside `idx['actions']`, but we have some cases where the action is actually inside `idx['neededToProceed']`

IIRC see we could have 3 scenarios

- submit a form by calling handleFormSave with some model params to be posted 
- invokeAction where the action is within idx['actions'] and we can just call it 
- submit a form without any specific model params (which today is happening by calling invokeAction with form name)

We could move the last scenario into a separate method to keep them separate?  That would mean we need to update the places where we are calling invokeAction with a formName to instead call this new method.

Reply:
for #3, I'm thinking we could refactor the `switchForm` function and replace `invokeAction` with `switchForm` event.